src/util/base58.rs

```diff
@@ -131,7 +131,7 @@ pub fn from(data: &str) -> Result<Vec<u8>, Error> {
     // Build in base 256
     for d58 in data.bytes() {
         // Compute "X = X * 58 + next_digit" in base 256
-        if d58 as usize > BASE58_DIGITS.len() {
+        if d58 as usize >= BASE58_DIGITS.len() {
```

Comment:
This is the right logic because we later do `BASE58_DIGITS[d58]` which would panic if `d58==BASE58_DIGITS.len()` *But* this can't really happen in practice, because `BASE58_DIGITS.len() == 128` and `data` is `&str` meaning it only contains valid UTF-8. and UTF-8 don't have any valid byte starting with 128. (it's either a single byte < 128 or multiple bytes with the first one > 128)

Reply:
Good catch